ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js

```diff
@@ -37,75 +46,117 @@ export default function UpdateSnapPermissionList({
   );
 
   const snapsMetadata = useSelector(getSnapsMetadata);
-  const snapsNameGetter = getSnapName(snapsMetadata);
 
-  const approvedPermissionsToShow = {
+  const approvedCombinedPermissions = {
     ...approvedPermissions,
     connection_permission: approvedConnections ?? {},
   };
 
-  const revokedPermissionsToShow = {
+  const revokedCombinedPermissions = {
     ...revokedPermissions,
     connection_permission: revokedConnections ?? {},
   };
 
-  const newPermissionsToShow = {
+  const newCombinedPermissions = {
     ...newPermissions,
     connection_permission: newConnections ?? {},
   };
 
+  const newWeightedPermissions = getWeightedPermissions({
+    t,
+    permissions: newCombinedPermissions,
+    subjectName: snapName,
+    getSubjectName: getSnapName(snapsMetadata),
+  });
+
+  const revokedWeightedPermissions = getWeightedPermissions({
+    t,
+    permissions: revokedCombinedPermissions,
+    subjectName: snapName,
+    getSubjectName: getSnapName(snapsMetadata),
+  });
+
+  const approvedWeightedPermissions = getWeightedPermissions({
+    t,
+    permissions: approvedCombinedPermissions,
+    subjectName: snapName,
+    getSubjectName: getSnapName(snapsMetadata),
+  });
+
+  const [showAll, setShowAll] = useState(
+    Object.keys(approvedWeightedPermissions).length < 1,
+  );
+
+  // Because approved permissions are sometimes hidden following the abstraction logic,
+  // it is needed sometimes to fill the gap in permission display, in certain edge cases
+  // when there is not enough new and revoked permissions to be shown.
+  let minApprovedPermissionsToShow;
+  const totalNewAndRevokedPermissions =
+    newWeightedPermissions.length + revokedWeightedPermissions.length;
+
+  switch (totalNewAndRevokedPermissions) {
+    case 0:
+      minApprovedPermissionsToShow = 3;
+      break;
+    case 1:
+      minApprovedPermissionsToShow = 2;
+      break;
+    case 2:
+      minApprovedPermissionsToShow = 1;
+      break;
+    default:
+      minApprovedPermissionsToShow = 0;
+  }
```

Comment:
```suggestion
  const totalNewAndRevokedPermissions =
    newWeightedPermissions.length + revokedWeightedPermissions.length;
    
   const minApprovedPermissionsToShow = Math.max(totalNewAndRevokedPermissions - MinPermissionAbstractionDisplayCount, 0);
```

Perhaps simpler like this?

Reply:
Yea, it can work, similar to that, but only if I reverse the subtraction part like this:
```MinPermissionAbstractionDisplayCount - totalNewAndRevokedPermissions```
That's what you wanted probably?

Reply:
Applied proposed code with alteration I mentioned above.